core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1068,6 +1096,104 @@ public void run() {
+              "maxBackoff": string,       // Long decimal with "s" appended
+              "backoffMultiplier": number
+              "retryableStatusCodes": []
+            }
+          }
+        ]
+      }
+      */
+
+      if (methodConfigs != null) {
+        for (JsonElement methodConfig : methodConfigs) {
+          JsonArray names = methodConfig.getAsJsonObject().getAsJsonArray("name");
+          JsonObject retryPolicy = methodConfig.getAsJsonObject().getAsJsonObject("retryPolicy");
+          if (retryPolicy != null) {
+            int maxAttempts = retryPolicy.getAsJsonPrimitive("maxAttempts").getAsInt();
+            String initialBackoffStr =
+                retryPolicy.getAsJsonPrimitive("initialBackoff").getAsString();
+            checkState(
+                initialBackoffStr.charAt(initialBackoffStr.length() - 1) == 's',
+                "invalid value of initialBackoff");
+            double initialBackoff =
+                Double.parseDouble(initialBackoffStr.substring(0, initialBackoffStr.length() - 1));
+            String maxBackoffStr =
+                retryPolicy.getAsJsonPrimitive("maxBackoff").getAsString();
+            checkState(
+                maxBackoffStr.charAt(maxBackoffStr.length() - 1) == 's',
+                "invalid value of maxBackoff");
+            double maxBackoff =
+                Double.parseDouble(maxBackoffStr.substring(0, maxBackoffStr.length() - 1));
+            double backoffMultiplier =
+                retryPolicy.getAsJsonPrimitive("backoffMultiplier").getAsDouble();
+            JsonArray retryableStatusCodes = retryPolicy.getAsJsonArray("retryableStatusCodes");
+            Set<Code> codeSet = new HashSet<Code>(retryableStatusCodes.size());
+            for (JsonElement retryableStatusCode : retryableStatusCodes) {
+              codeSet.add(Code.valueOf(retryableStatusCode.getAsString()));
+            }
+            RetryPolicy pojoPolicy = new RetryPolicy(
+                maxAttempts, initialBackoff, maxBackoff, backoffMultiplier, codeSet);
+
+            for (JsonElement name : names) {
+              String service = name.getAsJsonObject().getAsJsonPrimitive("service").getAsString();
+              JsonPrimitive method = name.getAsJsonObject().getAsJsonPrimitive("method");
+              if (method != null && !method.getAsString().isEmpty()) {
+                fullMethodNameMap.put(
+                    MethodDescriptor.generateFullMethodName(service, method.getAsString()),
+                    pojoPolicy);
+              } else {
+                serviceNameMap.put(service, pojoPolicy);
+              }
+            }
+          }
+        }
+      }
+    }
+
+    return new RetryPolicies() {
+      @Override
+      public RetryPolicy get(MethodDescriptor<?, ?> method) {
+        RetryPolicy retryPolicy = fullMethodNameMap.get(method.getFullMethodName());
+        if (retryPolicy == null) {
+          retryPolicy = serviceNameMap
```

Comment:
Add test coverage for this fallback logic?

Reply:
I think this test code covers it

```java
assertEquals(
          new RetryPolicy(
              3, parseDouble("2.1"), parseDouble("2.2"), parseDouble("3"),
              Arrays.asList(Code.UNAVAILABLE, Code.RESOURCE_EXHAUSTED)),
          retryPolicies.get(builder.setFullMethodName("SimpleService1/not_exist").build()));
```